superset-frontend/src/explore/actions/saveModalActions.js

```diff
@@ -43,6 +43,14 @@ export function fetchDashboards(userId) {
           value: id,
           label: (json.result[index] || {}).dashboard_title,
         }));
+        choices.sort((a, b) => {
+          const labelA = a.label.toUpperCase();
+          const labelB = b.label.toUpperCase();
+          return labelA.localeCompare(labelB, {
+            sensitivity: 'base',
+            numeric: true,
+          });
+        });
```

Comment:
[sensitivity: 'base'](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator#syntax) already handles case sensitivity.
```suggestion
        choices.sort((a, b) =>
          a.label.localeCompare(b.label, {
            sensitivity: 'base',
            numeric: true,
          }),
        );
```